diff -u cabal-install-0.14.0.orig/bootstrap.sh cabal-install-0.14.0/bootstrap.sh
--- cabal-install-0.14.0.orig/bootstrap.sh	2014-09-17 23:12:23.342655189 +0900
+++ cabal-install-0.14.0/bootstrap.sh	2014-09-17 23:12:38.942655057 +0900
@@ -119,7 +119,7 @@
   URL=${HACKAGE_URL}/${PKG}/${VER}/${PKG}-${VER}.tar.gz
   if which ${CURL} > /dev/null
   then
-    ${CURL} --fail -C - -O ${URL} || die "Failed to download ${PKG}."
+    ${CURL} -L --fail -O ${URL} || die "Failed to download ${PKG}."
   elif which ${WGET} > /dev/null
   then
     ${WGET} -c ${URL} || die "Failed to download ${PKG}."
