diff -u cabal-install-0.8.2.orig/bootstrap.sh cabal-install-0.8.2/bootstrap.sh
--- cabal-install-0.8.2.orig/bootstrap.sh	2014-09-17 23:07:02.634657901 +0900
+++ cabal-install-0.8.2/bootstrap.sh	2014-09-17 23:09:13.066656798 +0900
@@ -16,6 +16,7 @@
 GHC_PKG=${GHC_PKG:-ghc-pkg}
 WGET=${WGET:-wget}
 CURL=${CURL:-curl}
+FETCH=${FETCH:-fetch}
 TAR=${TAR:-tar}
 GUNZIP=${GUNZIP:-gunzip}
 SCOPE_OF_INSTALLATION="--user"
@@ -129,12 +130,15 @@
   URL=${HACKAGE_URL}/${PKG}/${VER}/${PKG}-${VER}.tar.gz
   if which ${CURL} > /dev/null
   then
-    ${CURL} -C - -O ${URL} || die "Failed to download ${PKG}."
+    ${CURL} -L --fail -O ${URL} || die "Failed to download ${PKG}."
   elif which ${WGET} > /dev/null
   then
     ${WGET} -c ${URL} || die "Failed to download ${PKG}."
+  elif which ${FETCH} > /dev/null
+  then
+    ${FETCH} ${URL} || die "Failed to download ${PKG}."
   else
-    die "Failed to find a downloader. 'wget' or 'curl' is required."
+    die "Failed to find a downloader. 'curl', 'wget' or 'fetch' is required."
   fi
   [ -f "${PKG}-${VER}.tar.gz" ] \
     || die "Downloading ${URL} did not create ${PKG}-${VER}.tar.gz"
