diff -u cabal-install-0.10.2.orig/bootstrap.sh cabal-install-0.10.2/bootstrap.sh
--- cabal-install-0.10.2.orig/bootstrap.sh	2014-09-17 23:10:42.254656044 +0900
+++ cabal-install-0.10.2/bootstrap.sh	2014-09-17 23:11:21.746655710 +0900
@@ -115,7 +115,7 @@
   URL=${HACKAGE_URL}/${PKG}/${VER}/${PKG}-${VER}.tar.gz
   if which ${CURL} > /dev/null
   then
-    ${CURL} --fail -C - -O ${URL} || die "Failed to download ${PKG}."
+    ${CURL} -L --fail -O ${URL} || die "Failed to download ${PKG}."
   elif which ${WGET} > /dev/null
   then
     ${WGET} -c ${URL} || die "Failed to download ${PKG}."
